unzip_http.py

```diff
@@ -59,36 +60,40 @@ class RemoteZipFile:
     def __init__(self, url):
         self.url = url
         self.http = urllib3.PoolManager()
+        self.zip_size = 0
+
+        # remote reading all files
+        self.zip_entries = {r.filename: r for r in self.infolist()}
```

Comment:
`infolist` makes network requests so we shouldn't do this in the constructor.  This is why I made it a property.  It should really be a `functools.cached_property` which I think is what you're getting after.

Adding `zip_size` as a member is +0 for me, if you find yourself using it then we can certainly add it.